Compilation error when target's `config.h` defines `USE_LED_STRIP`.

### Describe the bug

 I'm getting compile errors when defining `USE_LED_STRIP` in a target's `config.h` file.

`
./src/main/io/ledstrip.c:1284:8: error: unknown type name 'ledProfileSequence_t'
 1284 | static ledProfileSequence_t applySimpleProfile(timeUs_t currentTimeUs)
      |        ^~~~~~~~~~~~~~~~~~~
`


### To Reproduce

1) add the following line to a target's `config.h` file.

`#define USE_LED_STRIP` 

2) build the target locally, using `make <TARGET>`

### Expected behavior

Successful compilation.

### Support ID

```txt
N/A
```


### Flight controller

Any

### Other components

_No response_

### How are the different components wired up (including port information)

_No response_

### Add any other context about the problem that you think might be relevant here

It seems that at the time `common_pre.h` is included, when defining `USE_LED_STRIP` in a target.h config, that `USE_LED_STRIP` is NOT defined, and thus the check for `USE_LED_STRIP` does not evaluate to true and thus `USE_LED_STRIP_STATUS_MODE` is not defined by the code on line 281, which results in a compile error.

So it appears, that if target authors want to enable `LED_STRIP` in their target's config.h which will be the case for many FCs that include either a) a dedicated pin for LED strips, or b) on-board RGB leds, that the current workaround is to define both `USE_LED_STRIP` and `USE_LED_STRIP_STATUS_MODE` in the target's config.h which seems wrong.

Note that I'm not using any flags on the command line, I'm not doing a 'cloud' build or a 'core' build, I'm just doing a regular build. (with reference to https://github.com/betaflight/betaflight.com/issues/389#issuecomment-1984064807)

--- a/src/main/platform.h
+++ b/src/main/platform.h
@@ -26,11 +26,11 @@
 #pragma GCC poison sprintf snprintf
 #endif
 
-#include "target/common_pre.h"
-
 #ifdef USE_CONFIG
 #include "config.h"
 #endif
+
+#include "target/common_pre.h"
 
 // MCU specific platform from drivers/XX
 #include "platform_mcu.h"

